tools/mksurfdata_esmf/gen_mksurfdata_jobscript_single.py

```diff
@@ -81,15 +90,26 @@ def main ():
         runfile.write(f"#PBS -A {account} \n")
         runfile.write('#PBS -N mksurfdata \n')
         runfile.write('#PBS -j oe \n')
-        runfile.write('#PBS -q regular \n')
         runfile.write('#PBS -l walltime=30:00 \n')
-        runfile.write(f"#PBS -l select={number_of_nodes}:ncpus=36:mpiprocs={tasks_per_node} \n")
+        if machine == 'cheyenne':
+            runfile.write('#PBS -q regular \n')
+            runfile.write(f"#PBS -l select={number_of_nodes}:ncpus=36:mpiprocs={tasks_per_node} \n")
+        elif machine == 'casper':
+            runfile.write('#PBS -q casper \n')
+            runfile.write(f"#PBS -l select={number_of_nodes}:ncpus=12:mpiprocs={tasks_per_node} \n")
```

Comment:
You should add a check that only the allowed machine names can be set.

Reply:
As noted above since you can only pick a valid choice, the else isn't needed here.